hail/python/hailtop/aiocloud/aioazure/fs.py

```diff
@@ -398,10 +450,10 @@ async def isfile(self, url: str) -> bool:
         return await self.get_blob_client(url).exists()
 
     async def isdir(self, url: str) -> bool:
-        _, _, name = self.get_account_container_and_name(url)
-        assert not name or name.endswith('/'), name
+        _, _, name, _ = self.get_url_parts(url)
+        dir_name = name if not name or name.endswith('/') else name + '/'
```

Comment:
huh, what was here before looks like a bad assert. Is this a general bugfix or did you run into this specifically because of sas tokens?

Reply:
seemed like a bug - I hit it while doing some ad hoc testing